DSpace/DSpace
# Make bulk edit compatible with controlled vocabularies

## References
Fixes #9749 

## Description
This PR implements the solution proposed in #9749
- Consider the column header when determining whether the column is authority-controlled, instead of just the `[...]` suffix
- Be more strict about the authority value/confidence format in order to support `::` within the plain text value of a cell
    - If you provide an authority value, you must provide a confidence as well. Otherwise, the entire cell is considered to be the plain text value
    - Only the last two `::`s are considered to be authority/confidence separators, any previous `::`s are considered to be part of the plain text value

## Instructions for Reviewers
Confirm that this PR addresses the described issue without introducing others.
**Please consider whether it is sensible to alter existing CSV import behaviour in this way.**

### Testing
1. Set up a local DSpace REST instance on the `main` branch
2. Update the configuration to make the `dc.subject` authority-controlled wrt. the SRSC vocabulary, like in the IT configuration [here](https://github.com/ybnd/dspace/blob/bulk-edit-authority-issues/dspace-api/src/test/java/org/dspace/app/bulkedit/MetadataEditControlledVocabularyIT.java#L70-L82)
3. Deposit an Item with valid SRSC `dc.subject` values, including ones that are at least two "hierarchy levels deep"
    - These metadata will have `::` separators within the plain text values, as well as authority and confidence values
4. Export this Item using `metadata-export` and download the resulting CSV file
    - You will see that `::`s are used as both the hierarchy and authority separator
5. Make  afew (dummy) changes to the CSV file and re-import it using `metadata-import`
    - **The import will succeed, but the `dc.subject` values will be mangled**
6. Check out this PR's branch and rebuild DSpace
7. Run `metadata-import` on the same CSV
    - **The import will succeed, now the `dc.subject` values will be imported correctly**


## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
